src/library/scala/collection/mutable/ArrayOps.scala

```diff
@@ -296,6 +335,12 @@ final class ofUnit(override val repr: Array[Unit]) extends AnyVal with ArrayOps[
     override protected[this] def thisCollection: WrappedArray[Unit] = new WrappedArray.ofUnit(repr)
     override protected[this] def toCollection(repr: Array[Unit]): WrappedArray[Unit] = new WrappedArray.ofUnit(repr)
     override protected[this] def newBuilder = new ArrayBuilder.ofUnit
+    protected override def emptyImpl = emptyUnitArray
+    protected override def sliceImpl(from: Int, until: Int) = {
+      // cant use util.Arrays.copyOfRange[Unit](repr, from, until) - Unit is special and doesnt compile
+      // but Unit and null are analogous so values are not important to copy across
+      Array[Unit](until-from)
```

Comment:
I think you missed the `new`.  As written, this discards the value `until-from` to produce a single unit, then creates an array of length one.  There should be tests of all the slices for all the different implementations to catch stuff like this!